## Procedencia del dataset inmobiliario

- Título del dataset: Idealista MADRID
- Plataforma: Kaggle
- Autor de la publicación: fjcob1
- Fuente subyacente: anuncios de venta de Idealista
- Número de registros utilizados: 11.826
- Número de variables: 14
- Fecha exacta de extracción: no especificada
- Naturaleza temporal: fotografía histórica del mercado
- Archivo original del proyecto: Datos.csv
- Archivo conservado: idealista_listings_unknown_date.csv
- Fecha de verificación de la procedencia: 24/07/2026

##### Importar bibliotecas

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
pd.set_option("display.max_colwidth", 150)

In [2]:
from pathlib import Path
import pandas as pd

CURRENT_DIR = Path.cwd() #localización del directorio actual

if CURRENT_DIR.name == "notebooks":
    PROJECT_ROOT = CURRENT_DIR.parent

elif CURRENT_DIR.name == "TFG_Recomendador_Viviendas":
    PROJECT_ROOT = CURRENT_DIR

elif (
    CURRENT_DIR
    / "TFG_Recomendador_Viviendas"
).exists():
    PROJECT_ROOT = (
        CURRENT_DIR
        / "TFG_Recomendador_Viviendas"
    )

else:
    raise FileNotFoundError(
        "No se ha podido localizar la carpeta "
        "'TFG_Recomendador_Viviendas'.\n"
        f"Directorio actual: {CURRENT_DIR}"
    )


ruta_datos = (
    PROJECT_ROOT
    / "data"
    / "raw"
    / "historical"
    / "idealista_listings_unknown_date.csv"
)

print("Directorio actual:", CURRENT_DIR)
print("Raíz del proyecto:", PROJECT_ROOT)
print("Dataset:", ruta_datos)
print("¿Existe el archivo?", ruta_datos.exists())

if not ruta_datos.exists():
    raise FileNotFoundError(
        "No se encuentra el dataset en:\n"
        f"{ruta_datos}"
    )




Directorio actual: c:\Users\clave\OneDrive\Documentos\ASIGNATURAS 5º INF-ADE\TFG INFO\TFG_Recomendador_Viviendas\notebooks
Raíz del proyecto: c:\Users\clave\OneDrive\Documentos\ASIGNATURAS 5º INF-ADE\TFG INFO\TFG_Recomendador_Viviendas
Dataset: c:\Users\clave\OneDrive\Documentos\ASIGNATURAS 5º INF-ADE\TFG INFO\TFG_Recomendador_Viviendas\data\raw\historical\idealista_listings_unknown_date.csv
¿Existe el archivo? True


In [3]:
#Caraga del csv

df_raw = pd.read_csv(ruta_datos)

print("\nNúmero de filas:", df_raw.shape[0])
print("Número de columnas:", df_raw.shape[1])

display(df_raw.head()) #muestra los primeros 5 anuncios


Número de filas: 11826
Número de columnas: 14


,provincia,zona,titulo,PrecioActual,PrecioAnterior,metros,habitaciones,ascensor,localizacion,planta,baños,tags,descripcion,Enlace
0,madrid,ciudad-lineal,"Piso en calle de San Marcelo, 22, Ventas, Madrid",355000,0,69,2.0,S,EXTERIOR,5ª,1,"VIVIENDA,LUMINOSO,VISTAS,REFORMADA,PORTERO,EXTERIOR","Particular Vende vivienda totalmente reformada, cocina independiente, salón, dos dormitorios y un baño. Amplias vistas a zonas ajardinadas y porte...",https://www.idealista.com/inmueble/106956987/
1,madrid,carabanchel,"Piso en calle Cabo Nicolás Mur, San Isidro, Madrid",149000,159000,91,3.0,N,EXTERIOR,1ª,0,"PISO,EXCLUSIVA,INMOBILIARIA,OPORTUNIDAD","Inmobiliarias Encuentro vende en exclusiva la NUDA PROPIEDAD PARA INVERSORES Presentamos este fantástico piso en el barrio de San Isidro, Madrid, ...",https://www.idealista.com/inmueble/106906044/
2,madrid,centro,"Piso en calle de Rodas, Lavapiés-Embajadores, Madrid",195000,0,36,1.0,S,NaN,2ª,0,"EXCLUSIVA,ESTUDIO","ESTUDIO EN PLENO CENTRO DE MADRIDSarago Servicios Inmobiliarios presenta, en exclusiva, este fantástico estudio en segunda planta con ascensor, en...",https://www.idealista.com/inmueble/107306175/
3,madrid,usera,"Piso en calle de Ferroviarios, Almendrales, Madrid",195000,0,58,1.0,S,INTERIOR,BAJO,0,"VIVIENDA,HOGAR,FUNCIONAL","Esta acogedora vivienda, ubicada en una planta baja, es perfecta para quienes buscan un hogar listo para entrar a vivir, sin necesidad de realizar...",https://www.idealista.com/inmueble/106325171/
4,madrid,tetuan,"Dúplex en Bellas Vistas, Madrid",715000,750000,140,3.0,S,EXTERIOR,2ª,0,"TERRAZA,EXCLUSIVA,MODERNO","Maravilloso ATICO de reciente construcción, con TERRAZA en el centro de Madrid.En una calle muy tranquila, y RESIDENCIAL, cerca de todos los comer...",https://www.idealista.com/inmueble/106627265/


In [4]:
df_raw.info() #descripcion de la estructura del dataset, incluyendo el tipo de datos y la cantidad de valores no nulos en cada columna

<class 'pandas.DataFrame'>
RangeIndex: 11826 entries, 0 to 11825
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   provincia       11826 non-null  str    
 1   zona            11826 non-null  str    
 2   titulo          11826 non-null  str    
 3   PrecioActual    11826 non-null  int64  
 4   PrecioAnterior  11826 non-null  int64  
 5   metros          11826 non-null  int64  
 6   habitaciones    11460 non-null  float64
 7   ascensor        11033 non-null  str    
 8   localizacion    10730 non-null  str    
 9   planta          10601 non-null  str    
 10  baños           11826 non-null  int64  
 11  tags            11664 non-null  str    
 12  descripcion     11761 non-null  str    
 13  Enlace          11826 non-null  str    
dtypes: float64(1), int64(4), str(9)
memory usage: 6.7 MB


In [5]:
resumen_columnas = pd.DataFrame({
    "tipo": df_raw.dtypes.astype(str),
    "valores_no_nulos": df_raw.notna().sum(), #NaN  
    "valores_nulos": df_raw.isna().sum(),
    "porcentaje_nulos": (df_raw.isna().mean() * 100).round(2),
    "valores_unicos": df_raw.nunique() #demuestra que cada nunico tiene su propio enlace
})

display(resumen_columnas)

,tipo,valores_no_nulos,valores_nulos,porcentaje_nulos,valores_unicos
provincia,str,11826,0,0.00,1
zona,str,11826,0,0.00,21
titulo,str,11826,0,0.00,5499
PrecioActual,int64,11826,0,0.00,1880
PrecioAnterior,int64,11826,0,0.00,468
metros,int64,11826,0,0.00,615
habitaciones,float64,11460,366,3.09,17
ascensor,str,11033,793,6.71,2
localizacion,str,10730,1096,9.27,2
planta,str,10601,1225,10.36,27


In [6]:
# Valores mas frecuentes de cada columna
for columna in df_raw.columns:
    print(f"\n--- {columna} ---")
    print(df_raw[columna].value_counts(dropna=False).head(10))


--- provincia ---
provincia
madrid    11826
Name: count, dtype: int64

--- zona ---
zona
barrio-de-salamanca    2018
centro                 1890
chamberi                796
chamartin               706
tetuan                  596
moncloa                 580
carabanchel             567
ciudad-lineal           537
puente-de-vallecas      526
retiro                  508
Name: count, dtype: int64

--- titulo ---
titulo
Piso en Goya, Madrid                    203
Piso en Recoletos, Madrid               192
Piso en Castellana, Madrid              135
Piso en Malasaña-Universidad, Madrid    103
Piso en Lista, Madrid                   102
Piso en Chueca-Justicia, Madrid          88
Piso en Almagro, Madrid                  87
Piso en Palacio, Madrid                  81
Piso en El Viso, Madrid                  80
Piso en Sol, Madrid                      76
Name: count, dtype: int64

--- PrecioActual ---
PrecioActual
1200000    91
250000     87
450000     85
950000     80
1250000    75
750000    

#### CONCLUSIONES
- Vemos que todos los anuncios son de madrid y por lo tanto *proviencia* no sirve para diferenciar barrios dentro del dataser

- Existen 21 zonas generales *(corresponden aproximadamente al nivel de distrito, no al nivel de barrios -> Hay que extraer del título la subzona o barrio concreto y relacionarlo con la tabla territorial oficial)*

- Aparece el valor 0 en 10.713 anuncios, aproximadamente el 90,6 % *(PrecioAnterior = 0 significa que no se dispone de un precio anterior, es un ejemplo de valor ausente codificado como cero)*

- 9.311 anuncios tienen baños = 0, es decir, un 78,73 % del dataset *(No es razonable pensar que casi ocho de cada diez viviendas no tienen baño -> la columna original presenta un problema de extracción. Más adelante de buscara info en el titutlo, las etiquetas y la descripcion)*

- Ascensor solo tiene un 6,71% de ausente y localizacion un 9,27%

- Planta no es una columna numerica limpia, hay que transformar los datos a un criterio de formato comun.



In [7]:
display(
    df_raw.describe(
        include="all",
        percentiles=[0.01, 0.05, 0.25, 0.50, 0.75, 0.95, 0.99]
    ).T
)

,count,unique,top,freq,mean,std,min,1%,5%,25%,50%,75%,95%,99%,max
provincia,11826,1,madrid,11826,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
zona,11826,21,barrio-de-salamanca,2018,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
titulo,11826,5499,"Piso en Goya, Madrid",203,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
PrecioActual,11826.0,NaN,NaN,NaN,1030501.460173,1237717.630328,12000.0,86075.0,148625.0,289000.0,620000.0,1329000.0,3300000.0,6000000.0,23000000.0
PrecioAnterior,11826.0,NaN,NaN,NaN,73597.039151,363975.260549,0.0,0.0,0.0,0.0,0.0,0.0,420000.0,1598750.0,8450000.0
metros,11826.0,NaN,NaN,NaN,153.790039,766.21775,11.0,26.0,40.0,68.0,103.0,160.0,388.75,750.0,75000.0
habitaciones,11460.0,NaN,NaN,NaN,2.847731,1.432402,1.0,1.0,1.0,2.0,3.0,3.0,5.0,7.0,20.0
ascensor,11033,2,S,8419,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
localizacion,10730,2,EXTERIOR,9087,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
planta,10601,27,1ª,2108,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


#### **CONCLUSIONES**

- **PRECION ACTUAL**: La media es de 1.030.501€ mientras que la mediana es de 620.000€. ESto se debe a que hay viviendas con precios muy altos que hacen que la media aumente y no represente correctamente el precio habitual de lso precios. Por ello es mas adecuado usar la *mediana*

- **PRECIO ANTERIOR**: La mayoria de valores de esta columna es 0, lo que significa que no disponemos de esta información y estos ceros deben tratarse como datos ausente, no como precios reales.

- **SUPERFICIE**: La mediana es de 103m^2 muetras que la media es de 154 m^2. De neuvo se ve afectada la media porque existen viviendas con superficies muy grandes. Hay valores que tienen que revisarse porque pueden ser errores, terrenos, edificios completos o promociones inmoviliarias. 

- **NUMERO DE HABITACIONES**: La mayoria de viviendas tiene alrededor de 3 habitaciones, si embargo existen anucios con hasta 20 habitaciones, por lo que hay que comprobar si se corresponden con viviendas reales de gran tamaño o a otro tipo de inmueble.

- **NUMERO DE BAÑOS**: Esta columna presenta un problema importante porque la mayoria de los anuncios tiene un valor de 0. No es razonable que tantas viviendas no tengan baño, por lo que lo más probable es que esta información no hay sido recogida correctamente. Como se ha comentado anteiormente, se busacra informacion complementaria en etiquetas, titulos y descripciones de lso anuncios.

- **ZONAS DE MADRID**: Este dataset contiene 21 zonas, que se ocrresponden con areas generales de Madrid. Sin embargo, el recomendador usara los 131 barrios oficiales para obtener resultados mucho mas concretos. Por tanto, mas adelante de identificara un barrio por anuncion a partir de su informacion.

- **ASCENSOR, LOCACLIZACION Y PLANTA**: La mayoría de las viviendas indican que tienen ascensor y que son exteriores.in embargo, estas columnas también contienen valores ausentes y diferentes formas de escribir la información, especialmente en el caso de la planta (1ª, BAJO, ENTREPLANTA, SÓTANO, valores negativos), por ello se transformara estos datos para que todos sigan el mismo formato.

- **ANUNCIOS DUPLICADOS**: No se han encontrado filas completamente repetidas ni enlaces duplicados, ya que cada anuncio tiene un enlace diferente. Aun así, podría ocurrir que una misma vivienda aparezca publicada varias veces con enlaces o descripciones diferentes, por lo que más adelante se hará una revisión más completa de posibles duplicados.

**NOTA: ¿Qué son los percentiles?**
EJEMPLO: El percentil 95 de los metros es aproximadamente 388,75 m².

Significa: El 95 % de los anuncios tiene una superficie igual o inferior a 388,75 m².

Los percentiles ayudan a detectar valores extremos sin asumir automáticamente que son errores.

In [8]:
print("Número de zonas:", df_raw["zona"].nunique())

display(
    df_raw["zona"]
    .value_counts(dropna=False)
    .rename_axis("zona")
    .reset_index(name="numero_anuncios")
)

Número de zonas: 21


,zona,numero_anuncios
0,barrio-de-salamanca,2018
1,centro,1890
2,chamberi,796
3,chamartin,706
4,tetuan,596
5,moncloa,580
6,carabanchel,567
7,ciudad-lineal,537
8,puente-de-vallecas,526
9,retiro,508


En pirnciop el dataser ofrece cobertura de toda la ciudad. Sin embargo, el número de anuncios noe sta repartido de forma equilibrada.Solo Barrio de Salamanca y Centro reunen aproximadamente un tercio de todo los anuncios del dataset, en cambio las zonas con menos presencia son moratalaz y Barajas con solo 97 anuncios cada una. Esto significa que algunas zonas están mucho mejor representadas que otras y por tanto los anuncios de las zonas con muchos anuncios seran, en general, mas fiables que aquellas con una muestra pequeña.

In [9]:
columnas_a_revisar = [
    "provincia",
    "ascensor",
    "localizacion",
    "planta",
    "habitaciones",
    "baños"
]

for columna in columnas_a_revisar:
    print("\n" + "=" * 60)
    print(f"COLUMNA: {columna}")
    print("=" * 60)

    resumen = (
        df_raw[columna]
        .value_counts(dropna=False)
        .rename_axis("valor")
        .reset_index(name="numero_anuncios")
    )

    resumen["porcentaje"] = (
        resumen["numero_anuncios"] / len(df_raw) * 100
    ).round(2)

    display(resumen)


COLUMNA: provincia


,valor,numero_anuncios,porcentaje
0,madrid,11826,100.0



COLUMNA: ascensor


,valor,numero_anuncios,porcentaje
0,S,8419,71.19
1,N,2614,22.10
2,NaN,793,6.71



COLUMNA: localizacion


,valor,numero_anuncios,porcentaje
0,EXTERIOR,9087,76.84
1,INTERIOR,1643,13.89
2,NaN,1096,9.27



COLUMNA: planta


,valor,numero_anuncios,porcentaje
0,1ª,2108,17.83
1,2ª,1779,15.04
2,3ª,1570,13.28
3,BAJO,1509,12.76
4,4ª,1266,10.71
5,NaN,1225,10.36
6,5ª,816,6.90
7,6ª,574,4.85
8,7ª,289,2.44
9,ENTREPLANTA,163,1.38



COLUMNA: habitaciones


,valor,numero_anuncios,porcentaje
0,3.0,3768,31.86
1,2.0,3383,28.61
2,1.0,1624,13.73
3,4.0,1490,12.60
4,5.0,690,5.83
5,NaN,366,3.09
6,6.0,291,2.46
7,7.0,102,0.86
8,8.0,48,0.41
9,9.0,26,0.22



COLUMNA: baños


,valor,numero_anuncios,porcentaje
0,0,9311,78.73
1,1,1063,8.99
2,2,989,8.36
3,3,319,2.70
4,4,87,0.74
5,5,34,0.29
6,6,17,0.14
7,7,6,0.05


La diferencia con la revisión anterior es que aquí se muestra la distribución completa y no solo los diez valores más frecuentes. Todavía no decide qué valores son incorrectos, solo los señala para revisarlos posteriormente.

In [10]:
print("Filas completamente duplicadas:", df_raw.duplicated().sum())

print(
    "Enlaces duplicados:",
    df_raw["Enlace"].duplicated().sum()
)

Filas completamente duplicadas: 0
Enlaces duplicados: 0
